filters/auth/auth.go

```diff
@@ -133,12 +138,16 @@ func reject(
 	ctx.Serve(rsp)
 }
 
+func redirect(ctx filters.FilterContext, username string, reason rejectReason, destination, debuginfo string) {
+	reject(ctx, http.StatusFound, username, reason, "", debuginfo, destination)
```

Comment:
please provide the body of the function and not re-use `redirect`.
This will add a bit of code, but reduce the size of change.
`reject` should only be used for rejection as 4xx, thanks!